src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -92,17 +90,57 @@ CHIP_ERROR CASEServer::OnMessageReceived(Messaging::ExchangeContext * ec, const
     {
         Cleanup();
     }
+
     return err;
 }
 
-void CASEServer::Cleanup()
+void CASEServer::Cleanup(ScopedNodeId previouslyEstablishedPeer)
```

Comment:
Call this function `Cleanup` is quite contra-intuitive judging from what it does. Suggest rename this function to something like `PrepareForNextPairing`

Reply:
Given `Cleanup(peer)` is called on session error, why not evict of the old peer session right away here in `Cleanup` and remove the need to pass `previouslyEstablishedPeer` and `sessionEvictionHint` deeper into the call stack?  Those parameters become weirder (how do I pass "none") and less intuitive (what is this even for?) the deeper into the call stack they go.  Here in Cleanup, the parameter makes perfect sense -- this peer session errored out, so clean it up (evit it) and opportunistically listen for a new one or have upper layer client delegate initiate a new replacement one.

Reply:
Cleanup() is called on both error and on success. On error, we actually don't have a previous session.